### PR TITLE
[paymentservice] add basic metrics support

### PR DESCRIPTION
**Pull request by pichlermarc**
Related to #68

## Changes

This PR adds a counter `app.payment.transactions` that can be split by the transaction currency, adds dependencies to the JS Metrics packages, and changes the startup process so that the async resource detection can finish before creating a `Meter` or `Tracer`.

Once updated instrumentation packages are released, I will open a follow-up PR that updates the metrics API/SDK to the GA versions. Then the added temporality environment variable will also change the temporality of the exported metrics accordingly. :slightly_smiling_face: 

image

### CODE REVIEW

**Review by julianocosta89 — APPROVED** [review]
Nice one!
image

**Review by reyang — APPROVED** [review]

**Comment by cartersocha** [comment]
@pichlermarc would you mind updating our metric feature matrix too? Lgtm otherwise. Thanks for this!

https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/metric_service_features.md

**Comment by cartersocha** [comment]
Oh and we'll need to update the payment service doc in the docs/services folder

**Review by fatsheep9146 — APPROVED** [review]

**Comment by pichlermarc** [comment]
@cartersocha changed some text to the payment service docs to reflect the new setup, added some text for metrics, and changed the entry in the matrix :slightly_smiling_face:

File: docs/services/paymentservice.md [hunk 1]
```diff
@@ -8,33 +8,39 @@ processed.
...
 
-It is recommended to use a Node required module when starting your NodeJS
-application to initialize the SDK and auto-instrumentation. When initializing
-the OpenTelemetry NodeJS SDK, you optionally specify which auto-instrumentation
-libraries to leverage, or make use of the `getNodeAutoInstrumentations()`
-function which includes most popular frameworks. The `tracing.js` contains all
-code required to initialize the SDK and auto-instrumentation based on standard
-OpenTelemetry environment variables for OTLP export, resource attributes, and
-service name.
+It is recommended to use wrap your app using an OpenTelemetry initialization
```
Comment (mviitane) [inline docs/services/paymentservice.md]: The wrap concept could be explained a bit more. For example, for me, it's not clear if we use wrapping in the code below.

Comment (pichlermarc) [reply docs/services/paymentservice.md]: Tried to clarify the approach in [b626a9c](https://github.com/open-telemetry/opentelemetry-demo/pull/583/commits/b626a9c80d23b13a8b99be475e9093e4aa0c749b). :slightly_smiling_face:

**Review by mviitane — APPROVED** [review]

**Comment by cartersocha** [comment]
Thanks again @pichlermarc! Merging now

**Pull request merged**